azure-pipelines.yaml

```diff
@@ -57,40 +57,30 @@ steps:
     mkdir test-output
     chmod 777 test-output
   displayName: create test output directory
-  # build test image
+  # build base image
 - task: DockerCompose@0
   inputs:
+    action: Build services
     containerregistrytype: Azure Container Registry
     azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
     azureContainerRegistry: $(azureContainerRegistryFull)
     dockerComposeFile: docker-compose.yaml
     projectName: $(Build.Repository.Name)
     qualifyImageNames: true
-    dockerComposeCommand: -f docker-compose.test.yaml build --no-cache $(imageName)
-  displayName: build tests image
+    additionalImageTags: $(containerTag)
+  displayName: build base image
   # run tests
 - task: DockerCompose@0
   inputs:
     containerregistrytype: Azure Container Registry
     azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
     azureContainerRegistry: $(azureContainerRegistryFull)
     dockerComposeFile: docker-compose.yaml
+    additionalDockerComposeFiles: docker-compose.test.yaml
     projectName: $(Build.Repository.Name)
     qualifyImageNames: true
-    dockerComposeCommand: -f docker-compose.test.yaml up --force-recreate  $(imageName)
+    dockerComposeCommand: run $(imageName) npm run test
```

Comment:
rather than using run and specifing the command here it could be set in the `docker-compose.test.yaml`

Reply:
Fair point. This just came from iterating through other solutions before arriving back at the separate test compose file.

Reply:
I've changed this to `dockerComposeCommand: run $(imageName)` and set the command to `npm run test` in the Docker Compose file.